src/o365/spo/commands/apppage/apppage-add.ts

```diff
@@ -0,0 +1,153 @@
+import auth from '../../SpoAuth';
+import config from '../../../../config';
+import request from '../../../../request';
+import commands from '../../commands';
+import {
+  CommandOption, CommandValidate
+} from '../../../../Command';
+import SpoCommand from '../../SpoCommand';
+import { ContextInfo } from '../../spo';
+import GlobalOptions from '../../../../GlobalOptions';
+import { Auth } from '../../../../Auth';
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  webUrl: string;
+  title: string;
+  webPartData: string;
+  addToQuickLaunch: boolean;
+}
+
+class SpoAppPageAddCommand extends SpoCommand {
+  public get name(): string {
+    return `${commands.APPPAGE_ADD}`;
+  }
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.addToQuickLaunch = args.options.addToQuickLaunch;
+    return telemetryProps;
+  }
+  public get description(): string {
+    return 'Creates a single-part app page';
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: () => void): void {
+    const resource: string = Auth.getResourceFromUrl(args.options.webUrl);
+    let siteAccessToken: string = '';
+
+    auth
+      .getAccessToken(resource, auth.service.refreshToken as string, cmd, this.debug)
+      .then((accessToken: string): Promise<ContextInfo> => {
+        siteAccessToken = accessToken;
+
+        if (this.debug) {
+          cmd.log(`Retrieved access token ${accessToken}. Retrieving request digest...`);
+        }
+
+        if (this.verbose) {
+          cmd.log(`Retrieving request digest...`);
+        }
+
+        return this.getRequestDigestForSite(args.options.webUrl, siteAccessToken, cmd, this.debug);
+      })
+      .then((res: ContextInfo): Promise<{}> => {
+        const requestOptions: any = {
+          url: `${auth.site.url}/_api/sitepages/Pages/CreateFullPageApp`,
```

Comment:
Could you please change the `auth.site.url` with `args.options.webUrl` since it is creating app pages in the root site no matter what I have specified as webUrl?

Reply:
great pick up @VelinGeorgiev  :) thanks I will push these 3 things in asap, cheers